renderdoc/driver/gl/gl_manager.cpp

```diff
@@ -608,8 +609,22 @@ void GLResourceManager::PrepareTextureInitialContents(ResourceId liveid, Resourc
 
     if(HasExt[ARB_texture_swizzle] || HasExt[EXT_texture_swizzle])
     {
-      gl.glGetTextureParameterivEXT(res.name, details.curType, eGL_TEXTURE_SWIZZLE_RGBA,
-                                    (GLint *)&state->swizzle[0]);
+      if(IsGLES)
```

Comment:
Unless there's a driver bug setting them individually vs. all at once I don't think there's any gain by branching and doing the GL version. This isn't inside a tight loop, so it's better to have more clear code.

Yes just add a `GLHookSet &` parameter, you can add the function in `gl_resources.h` for example - see similar functions for `GetSizedFormat` or `GetFramebufferMipAndLayer`. You could also perhaps make it take `GLenum *swizzleRGBA` and move the cast inside, to make it a bit nicer.

It's worth moving to a utility function since it's a repeated pattern now rather than one function call before. With any codebase it's a good idea to try to leave the code better than you found it when possible. Just because there's other code which isn't clean doesn't mean you shouldn't try to organise new code or clean up while changing like this.